entities/src/test/java/org/odk/collect/entities/javarosa/filter/LocalEntitiesFilterStrategyTest.kt

```diff
@@ -541,6 +541,124 @@ class LocalEntitiesFilterStrategyTest {
         assertThat(fallthroughFilterStrategy.fellThrough, equalTo(false))
         assertThat(instanceProvider.fullParsePerformed, equalTo(false))
     }
+
+    @Test
+    fun `works correctly but not in the optimized way with question = expressions`() {
+        entitiesRepository.save("things", Entity.New("thing1", "Thing1"))
+
+        val scenario = Scenario.init(
+            "Secondary instance form",
+            html(
+                head(
+                    title("Secondary instance form"),
+                    model(
+                        mainInstance(
+                            t(
+                                "data id=\"create-entity-form\"",
+                                t("ref_question"),
+                                t("question1"),
+                                t("question2"),
+                                t("question3"),
+                            )
+                        ),
+                        t("instance id=\"things\" src=\"jr://file-csv/things.csv\""),
+                        bind("/data/ref_question").type("string"),
+                        bind("/data/question1").type("string"),
+                        bind("/data/question2").type("string"),
+                        bind("/data/question3").type("string")
+                    )
+                ),
+                body(
+                    input("/data/ref_question"),
+                    select1Dynamic(
+                        "/data/question1",
+                        "instance('things')/root/item[/data/ref_question='']",
+                        "name",
+                        "label"
+                    ),
+                    select1Dynamic(
+                        "/data/question2",
+                        "instance('things')/root/item[/data/ref_question=undefined]",
+                        "name",
+                        "label"
+                    ),
+                    select1Dynamic(
+                        "/data/question3",
+                        "instance('things')/root/item[/data/ref_question=null]",
```

Comment:
We don't need a case for `undefined` and `null` here as they're testing the same thing. Also can we make it clearer that we're testing against a non-existent child here by using `not_existing_property` instead of either like below?